test_semver.py

```diff
@@ -655,3 +656,43 @@ def test_should_process_raise_error(capsys):
     assert rc != 0
     captured = capsys.readouterr()
     assert captured.err.startswith("ERROR")
+
+
+@pytest.mark.parametrize("version,parts,expected", [
+    ("3.4.5", dict(major=2), '2.4.5'),
+    ("3.4.5", dict(major="2"), '2.4.5'),
+    ("3.4.5", dict(major=2, minor=5), '2.5.5'),
+    ("3.4.5", dict(minor=2), '3.2.5'),
+    ("3.4.5", dict(major=2, minor=5, patch=10), '2.5.10'),
+    ("3.4.5", dict(major=2, minor=5, patch=10, prerelease="rc1"),
+     '2.5.10-rc1'),
+    ("3.4.5", dict(major=2, minor=5, patch=10, prerelease="rc1", build="b1"),
+     '2.5.10-rc1+b1'),
+    ("3.4.5-alpha.1.2", dict(major=2), '2.4.5-alpha.1.2'),
+    ("3.4.5-alpha.1.2", dict(build="x1"), '3.4.5-alpha.1.2+x1'),
+    ("3.4.5+build1", dict(major=2), '2.4.5+build1'),
+])
+def test_replace_method_replaces_request_parts(version, parts, expected):
+    assert replace(version, **parts) == expected
+
+
+def test_should_raise_keyerror_for_invalid_key_in_replace():
+    with pytest.raises(TypeError):
+        assert replace("1.2.3", unknown="should_raise")
+
+
+@pytest.mark.parametrize("version,parts,expected", [
+    ("3.4.5", dict(major=2, minor=5), '2.5.5'),
+    ("3.4.5", dict(major=2, minor=5, patch=10), '2.5.10'),
+    ("3.4.5-alpha.1.2", dict(major=2), '2.4.5-alpha.1.2'),
+    ("3.4.5-alpha.1.2", dict(build="x1"), '3.4.5-alpha.1.2+x1'),
+    ("3.4.5+build1", dict(major=2), '2.4.5+build1'),
+])
+def test_should_versioninfo_replace(version, parts, expected):
```

Comment:
"versioninfo_replace_should_return_versioninfo_with_replaced_parts"

Reply:
@Flamefire 
Thanks, I don't mind changing that, but that makes the line **very** long. :wink: Additionally, you get a Flake8 E501 error ("line too long"). Could we have a shorter name?

By the way, have you worked with [GitHub Suggestions (step 6)](https://help.github.com/en/articles/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request)? It makes it a bit easier to work with feedback. You only click the "+/-" icon (the one between "Preview" and the "AA") and change the line to what you would like to have. It gives you a diff-like view.